mmf/models/mmbt.py

```diff
@@ -227,90 +230,70 @@ def forward(
         if attention_mask.dim() == 3:
-            extended_attention_mask = attention_mask[:, None, :, :]
+            attention_mask = attention_mask[:, None, :, :]
 
         # Provided a padding mask of dimensions [batch_size, seq_length]
         # - if the model is a decoder, apply a causal mask in addition to the
         #   padding mask
         # - if the model is an encoder, make the mask broadcastable to
         # [batch_size, num_heads, seq_length, seq_length]
         if attention_mask.dim() == 2:
-            if self.config.is_decoder:
+            if self.is_decoder:
                 batch_size, seq_length = input_shape
                 seq_ids = torch.arange(seq_length, device=device)
                 causal_mask = (
                     seq_ids[None, None, :].repeat(batch_size, seq_length, 1)
                     <= seq_ids[None, :, None]
                 )
-                extended_attention_mask = (
+                attention_mask = (
                     causal_mask[:, None, :, :] * attention_mask[:, None, None, :]
                 )
             else:
-                extended_attention_mask = attention_mask[:, None, None, :]
+                attention_mask = attention_mask[:, None, None, :]
 
         # Since attention_mask is 1.0 for positions we want to attend and 0.0 for
         # masked positions, this operation will create a tensor which is 0.0 for
         # positions we want to attend and -10000.0 for masked positions.
         # Since we are adding it to the raw scores before the softmax, this is
         # effectively the same as removing these entirely.
-        extended_attention_mask = extended_attention_mask.to(
-            dtype=next(self.parameters()).dtype
-        )  # fp16 compatibility
-        extended_attention_mask = (1.0 - extended_attention_mask) * -10000.0
+        # Python builtin next is currently not supported in Torchscript
+        if not torch.jit.is_scripting():
+            attention_mask = attention_mask.to(
+                dtype=next(self.parameters()).dtype
+            )  # fp16 compatibility
+        attention_mask = (1.0 - attention_mask) * -10000.0
 
         # If a 2D ou 3D attention mask is provided for the cross-attention
         # we need to make broadcastabe to
         # [batch_size, num_heads, seq_length, seq_length]
         if encoder_attention_mask.dim() == 3:
-            encoder_extended_attention_mask = encoder_attention_mask[:, None, :, :]
+            encoder_attention_mask = encoder_attention_mask[:, None, :, :]
         if encoder_attention_mask.dim() == 2:
-            encoder_extended_attention_mask = encoder_attention_mask[:, None, None, :]
-
-        encoder_extended_attention_mask = encoder_extended_attention_mask.to(
-            dtype=next(self.parameters()).dtype
-        )  # fp16 compatibility
-        encoder_extended_attention_mask = (
-            1.0 - encoder_extended_attention_mask
-        ) * -10000.0
-
-        # Prepare head mask if needed
```

Comment:
Removing this as we always set head_mask to None in the call to `MMBTModel`. Hence this is never used. Also scripting List[None] is not supported in the `BertEncoder`.